First of all - read this article carefully, not just skim it.

Try command line, instead of IPython.
In my case it doesn't work:

```
test@ce3b6ada1d29:~/beprof$ ls
LICENSE  README.md  beprof  setup.py
test@ce3b6ada1d29:~/beprof$ python3
Python 3.4.3 (default, Oct 14 2015, 20:28:29) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import beprof.curve as c
>>> c.main()

Subtract method :

c: 
 shape: (3, 2)
X : [0.000,10.000]
Y : [0.000000,5.000000]
Metadata : {}
d: 
 shape: (3, 2)
X : [0.000,10.000]
Y : [1.000000,1.000000]
Metadata : {}
c-d: 
 shape: (3, 2)
X : [0.000,10.000]
Y : [-1.000000,4.000000]
Metadata : {}
*********
X: [  0.   5.  10.]
Y: [-1.  4. -1.]
Newobj creation #1



a: 

X:  [0 1 2 3]
Y:  [0 1 2 1]
M:  {'positive': 'example'}

b: 

X:  [ 0.5  1.5  2.   2.5]
Y:  [ 1.  1.  1.  1.]
M:  {'negative': 'one'}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/test/beprof/beprof/curve.py", line 268, in main
    diff = functions.subtract(a, b)
  File "/home/test/beprof/beprof/functions.py", line 18, in subtract
    obj = curve1.__class__(np.stack((coord1, coord2), axis=1), **curve1.__dict__['metadata'])
AttributeError: 'module' object has no attribute 'stack'
```

It should be possible to run main method from one single command which I copy-paste into my bash shell. 

Hint: read about "python -m".

--- a/beprof/profile.py
+++ b/beprof/profile.py
@@ -20,6 +20,7 @@ class Profile(curve.Curve):
             new.axis = getattr(input_array,'axis',None)
         else:
             new.axis = axis
+        return new
 
     def __array_finalize__(self, obj):
         # print("Here I am in Profile.__array_finalize__ obj: ", type(obj))
@@ -210,6 +211,8 @@ class DepthProfile(Profile):
 def main():
     print('\nProfile')
     p = Profile([[0, 0], [1, 1], [2, 2], [3, 1]], some='exemplary', meta='data')
+    print(p)
+    print(type(p))
     print("X:", p.x)
     print("Y:", p.y)
     print('M: ', p.metadata)

